cmd/main_test.go

```diff
@@ -301,3 +304,60 @@ var _ = DescribeTable("TestRunAccessCheck", func(user string, namespace string,
 		"patch",
 		false),
 )
+
+var _ = DescribeTable("TestGetUserNamespaces",
+	func(labelKey string, labelValues []string, expectedNamespaces []string) {
+		e := echo.New()
+
+		req, err := labels.NewRequirement(labelKey, selection.In, labelValues)
+		Expect(err).NotTo(HaveOccurred(), "Error creating label requirement")
+
+		namespaces, err := getUserNamespaces(e, *req)
+		Expect(err).NotTo(HaveOccurred(), "Error getting user namespaces")
+
+		var actualNamespaces []string
+		for _, ns := range namespaces {
+			actualNamespaces = append(actualNamespaces, ns.Name)
+		}
+
+		// Expected and actual namespaces
+		log.Printf("Expected Namespaces: %v, Actual Namespaces: %v", expectedNamespaces, actualNamespaces)
+
+		// Check if the actual namespaces match the expected namespaces
+		Expect(actualNamespaces).To(ConsistOf(expectedNamespaces))
+	},
+	Entry(
+		"Get all user namespaces",
+		"konflux.ci/type",
+		[]string{"user"},
+		[]string{"test-tenant", "test-tenant-2", "test-tenant-3"},
+	),
+	// This test checks if we can retrieve all namespaces labeled as `user` namespaces.
+	// Expected output is it should return all three namespaces.
+
+	Entry(
+		"Get specific user namespace",
+		"kubernetes.io/metadata.name",
+		[]string{"test-tenant"},
+		[]string{"test-tenant"},
+	),
+	// This test checks if we can retrieve a single specific namespace by its name given.
+	// Expected is it should return only the "test-tenant" namespace.
+
+	Entry(
+		"Get multiple specific user namespaces",
+		"kubernetes.io/metadata.name",
+		[]string{"test-tenant", "test-tenant-2"},
+		[]string{"test-tenant", "test-tenant-2"},
+	),
+	// This test checks if we can retrieve multiple specific namespaces by their names given.
+	// Exepected is it should return both "test-tenant" and "test-tenant-2" namespaces.
```

Comment:
It should be understandable from the `expectedNamespaces` part of the Entry, no need to mention it.